examples/espressif/esp/src/stepper_driver.zig

```diff
@@ -0,0 +1,66 @@
+const std = @import("std");
+const microzig = @import("microzig");
+const hal = microzig.hal;
+const gpio = hal.gpio;
+const time = hal.time;
+
+const GPIO_Device = hal.drivers.GPIO_Device;
+const ClockDevice = hal.drivers.ClockDevice;
+const A4988 = microzig.drivers.stepper.A4988;
+
+const usb_serial_jtag = hal.usb_serial_jtag;
+
+pub const microzig_options = microzig.Options{
+    .logFn = usb_serial_jtag.logger.logFn,
+};
+
+pub fn main() !void {
+    var cd = ClockDevice{};
+
+    // Setup all pins for the stepper driver
+    var pins: struct {
+        ms1: GPIO_Device,
+        ms2: GPIO_Device,
+        ms3: GPIO_Device,
+        dir: GPIO_Device,
+        step: GPIO_Device,
+    } = undefined;
+    inline for (std.meta.fields(@TypeOf(pins)), .{ 0, 1, 2, 20, 10 }) |field, num| {
+        const pin = gpio.instance.num(num);
+        // Give the pin a sane default config
+        pin.apply(.{});
```

Comment:
Interesting that I had to do this. I guess the default values for a bunch of registers don't work. If that's the case, then maybe the set direction method should do more than just set outp.t